src/Vimeo/Vimeo.php

```diff
@@ -256,63 +256,97 @@ public function buildAuthorizationEndpoint($redirect_uri, $scope = 'public', $st
+    public function upload($file_path)
     {
         // Validate that our file is real.
         if (!is_file($file_path)) {
             throw new VimeoUploadException('Unable to locate file to upload.');
         }
 
-        // Begin the upload request by getting a ticket
-        $ticket_args = array('type' => 'streaming');
-        if ($machine_id !== null) {
-            $ticket_args['machine_id'] = $machine_id;
+        $file_size = filesize($file_path);
+
+        // If the user does not have enough free space in their quota to upload this, then don't.
+        $response = $this->request('/me', array('fields' => 'upload_quota.space.free'), 'GET');
+        if ($response['status'] !== 200) {
+            $error = !empty($response['body']['error']) ? ' [' . $response['body']['error'] . ']' : '';
+            throw new VimeoUploadException('Unable to pull the users upload quota.' . $error);
+        } elseif ($file_size > $response['body']['upload_quota']['space']['free']) {
+            throw new VimeoUploadException('User does not have any more free space to upload this video.');
+        }
+
+        // Begin the upload request by creating an attempt
+        $args = array(
+            'type' => 'tus',
+            'filename' => basename($file_path),
+            'size' => $file_size
+        );
+
+        // Use JSON filtering so we only receive the data that we need to make an upload happen.
+        $uri = '/me/videos?fields=uri,upload_link';
+        // @todo switch `upload_link` to `upload` to this when kenyas 2step PR is in prod.
+
+        $attempt = $this->request($uri, $args, 'POST');
+        if ($attempt['status'] !== 201) {
+            $attempt_error = !empty($attempt['body']['error']) ? ' [' . $attempt['body']['error'] . ']' : '';
+            throw new VimeoUploadException('Unable to initiate an upload attempt.' . $attempt_error);
         }
-        $ticket = $this->request('/me/videos', $ticket_args, 'POST');
 
-        return $this->perform_upload($file_path, $ticket);
+        return $this->perform_upload_tus($file_path, $file_size, $attempt);
     }
 
     /**
      * Replace the source of a single Vimeo video.
      *
+     * @link https://developer.vimeo.com/api/endpoints/videos#PUT/videos/{video_id}/files
      * @param string $video_uri Video uri of the video file to replace.
      * @param string $file_path Path to the video file to upload.
-     * @param string|null $machine_id
      * @throws VimeoUploadException
      * @return string Status
      */
-    public function replace($video_uri, $file_path, $machine_id = null)
+    public function replace($video_uri, $file_path)
     {
         //  Validate that our file is real.
         if (!is_file($file_path)) {
             throw new VimeoUploadException('Unable to locate file to upload.');
         }
 
+        // There's some complicated business logic surrounding upload quotas and replacing files so we can't check
```

Comment:
I don't believe this comment adds much to this public library

Reply:
Mostly left it in in case someone comes along to add quota checks to replace calls like others have done for the JS and Python libraries. I can pull it though.